mozetl/churn/schema.py

```diff
@@ -0,0 +1,51 @@
+from pyspark.sql.types import (DoubleType, LongType, StringType, StructField,
+                               StructType)
+
+"""
+- channel (appUpdateChannel)
+- geo (bucketed into top 30 countries + "rest of world")
+- is_funnelcake (contains "-cck-"?)
+- acquisition_period (cohort_week)
+- start_version (effective version on profile creation date)
+- sync_usage ("no", "single" or "multiple" devices)
+- current_version (current appVersion)
+- current_week (week)
+- source (associated attribution)
+- medium (associated with attribution)
+- campaign (associated with attribution)
+- content (associated with attribution)
+- distribution_id (funnelcake associated with profile)
+- default_search_engine
+- locale
+- is_active (were the client_ids active this week or not)
+- n_profiles (count of matching client_ids)
+- usage_hours (sum of the per-client subsession lengths,
+        clamped in the [0, MAX_SUBSESSION_LENGTH] range)
+- sum_squared_usage_hours (the sum of squares of the usage hours)
+- total_uri_count (sum of per-client uri counts)
+- unique_domains_count_per_profile (average of the average unique
+         domains per-client)
+"""
+churn_schema = StructType([
+    StructField('channel', StringType(), True),
```

Comment:
nit: Consider adding the above commentary in the "metadata" for each field

Reply:
This is interesting, I like it.